db/attachment_test.go

```diff
@@ -1425,63 +1433,101 @@ func TestAllowedAttachments(t *testing.T) {
 		})
 	}
 
-	// Multiple documents associated with multiple attachments of different digests.
+	// Multiple documents associated with multiple attachments with different digests.
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
 			ctx := &BlipSyncContext{}
 			meta := []AttachmentStorageMeta{
 				{digest: "digest1", version: tt.inputAttVersion},
 				{digest: "digest2", version: tt.inputAttVersion},
 			}
-			var (
-				docID1 = "doc1"
-				docID2 = "doc2"
-			)
+
+			docID1 := "doc1"
 			ctx.addAllowedAttachments(docID1, meta, tt.inputBlipProtocol)
 			requireIsAttachmentAllowedTrue(t, ctx, docID1, meta, tt.inputBlipProtocol)
+
+			docID2 := "doc2"
 			ctx.addAllowedAttachments(docID2, meta, tt.inputBlipProtocol)
 			requireIsAttachmentAllowedTrue(t, ctx, docID2, meta, tt.inputBlipProtocol)
 
 			ctx.removeAllowedAttachments(docID1, meta, tt.inputBlipProtocol)
-			if tt.inputBlipProtocol == BlipCBMobileReplicationV2 && tt.inputAttVersion == AttVersion1 {
-				requireIsAttachmentAllowedTrue(t, ctx, docID1, meta, tt.inputBlipProtocol)
-			} else {
+
+			if tt.inputBlipProtocol == BlipCBMobileReplicationV2 {
+				if tt.inputAttVersion == AttVersion1 {
+					requireIsAttachmentAllowedTrue(t, ctx, docID1, meta, tt.inputBlipProtocol)
+					requireIsAttachmentAllowedTrue(t, ctx, docID2, meta, tt.inputBlipProtocol)
+				} else if tt.inputAttVersion == AttVersion2 {
+					requireIsAttachmentAllowedFalse(t, ctx, docID1, meta, tt.inputBlipProtocol)
+					requireIsAttachmentAllowedFalse(t, ctx, docID2, meta, tt.inputBlipProtocol)
```

Comment:
At this point we haven't removed docID2, so isAttachmentAllowed should still be returning true.  This looks to be related to the question discussed earlier, about whether the counter should be incremented in the BlipCBMobileReplicationV2, AttVersion2 case.  Based on this I think the answer is yes - as long as we're using BlipCBMobileReplicationV2 there may be multiple in-flight revisions with matching allowedAttachments indexes.

Reply:
Yeah, that makes sense, changes are included to increment the counter when the protocol is BlipCBMobileReplicationV2 and attachment version is AttVersion2.